From 33cdfbe9d451caf03e2a0717404b548739727665 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Mon, 11 Oct 2021 16:10:42 -0400
Subject: [PATCH 18/23] Revert "Remove support for audio HAL V2 from the
 framework"

This reverts commit cbf517f837f7bf8a59f3ff8aa1e0e3e19612e251.

Also enable new extensions required to actually build

Change-Id: Icfd638a91b22ffa6d2cd6d19a624e699a85a850a
---
 media/libaudiohal/Android.bp         |  1 +
 media/libaudiohal/FactoryHalHidl.cpp |  1 +
 media/libaudiohal/impl/Android.bp    | 18 ++++++++++++++++++
 3 files changed, 20 insertions(+)

diff --git a/media/libaudiohal/Android.bp b/media/libaudiohal/Android.bp
index bd24c84dbc..f7a9ecc2b7 100644
--- a/media/libaudiohal/Android.bp
+++ b/media/libaudiohal/Android.bp
@@ -23,6 +23,7 @@ cc_library_shared {
     ],
 
     required: [
+        "libaudiohal@2.0",
         "libaudiohal@4.0",
         "libaudiohal@5.0",
         "libaudiohal@6.0",
diff --git a/media/libaudiohal/FactoryHalHidl.cpp b/media/libaudiohal/FactoryHalHidl.cpp
index e420d079b8..7228b221b9 100644
--- a/media/libaudiohal/FactoryHalHidl.cpp
+++ b/media/libaudiohal/FactoryHalHidl.cpp
@@ -35,6 +35,7 @@ const char* sAudioHALVersions[] = {
     "6.0",
     "5.0",
     "4.0",
+    "2.0",
     nullptr
 };
 
diff --git a/media/libaudiohal/impl/Android.bp b/media/libaudiohal/impl/Android.bp
index a2c6e8aeba..4e360be7e8 100644
--- a/media/libaudiohal/impl/Android.bp
+++ b/media/libaudiohal/impl/Android.bp
@@ -62,6 +62,24 @@ cc_defaults {
     ],
 }
 
+cc_library_shared {
+    name: "libaudiohal@2.0",
+    defaults: ["libaudiohal_default"],
+    shared_libs: [
+        "android.hardware.audio.common@2.0",
+        "android.hardware.audio.common@2.0-util",
+        "android.hardware.audio.effect@2.0",
+        "android.hardware.audio.effect@2.0-util",
+        "android.hardware.audio@2.0",
+        "android.hardware.audio@2.0-util",
+    ],
+    cflags: [
+        "-DMAJOR_VERSION=2",
+        "-DMINOR_VERSION=0",
+        "-include common/all-versions/VersionMacro.h",
+    ]
+}
+
 cc_library_shared {
     name: "libaudiohal@4.0",
     defaults: ["libaudiohal_default"],
-- 
2.25.1

